src/Http/Routing/test/UnitTests/Constraints/FIleNameRouteConstraintTest.cs

```diff
@@ -0,0 +1,100 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using Xunit;
+
+namespace Microsoft.AspNetCore.Routing.Constraints
+{
+    public class FileNameRouteConstraintTest
+    {
+        public static TheoryData<object> FileNameData
+        {
+            get
+            {
+                return new TheoryData<object>()
+                {
+                    "hello.txt",
+                    "hello.txt.jpg",
+                    "/hello.t",
+                    "/////hello.x",
+                    "a/b/c/d.e",
+                    "a/b./.c/d.e",
+                    ".gitnore",
+                    ".a",
+                    "/.......a"
+                };
+            }
+        }
+
+
+        [Theory]
+        [MemberData(nameof(FileNameData))]
+        public void Match_RouteValue_IsFileName(object value)
```

Comment:
Could you also add a unit test that specifically tests `FileNameRouteConstraint.IsFileName`? 

```C#
[Theory]
[MemberData(nameof(FileNameData))]
public void FileNameConstraint_IsFileNameReturnsTrue(string fileName)
{
   // Act
  var result = FileNameConstraint.IsFileName(fileName);

   // Assert
   Assert.True(result);
}
```

The method is internal and we don't necessarily need the ceremony of routing to verify that method.

Reply:
Same deal with the NonFileNameData

Reply:
I don't really get why testing the same thing twice will help.